Tensorflow Script for DCGAN (#18)
huggingface/community-events: huggan/tensorflow/dcgan/README.md

## Train DCGAN on your custom data
This folder contains a script to train DCGAN for unconditional image generation, leveraging the Hugging Face ecosystem for processing your data and pushing the model to the Hub.

> The script leverages 🤗 Datasets for loading and processing data, and TensorFlow for training the model and 🤗 Hub for hosting it.

Reviewer: ```suggestion
The script leverages 🤗 Datasets for loading and processing data, TensorFlow for training the model and 🤗 Hub for hosting it.
```

Reviewer 2: Actually grammar-wise the current one is already correct I think :smile:

Reviewer: Haha yeah sorry this is a major nitpick. I usually only include "and" at the end

## Launching the script

> You can simply run `python train.py` with default parameters. It will download MNIST dataset, preprocess it and train a model on it, will save results of epochs in a local directory and push to 🤗 Hub.

Reviewer: ```suggestion
You can simply run `python train.py` with the default parameters. It will download the MNIST dataset, preprocess it and train a model on it, will save results after each epoch in a local directory and push the model to the 🤗 Hub.
```

To train on another dataset available on the hub, simply do (for instance):

> python train.py --dataset cifar-10
> In case you'd like to tweak the script to your liking, first fork the "community-events" repo (see the button on the top right), then clone it locally:

Reviewer: ```suggestion
```bash
python train.py --dataset cifar-10
```

In case you'd like to tweak the script to your liking, first fork the "community-events" repo (see the button on the top right), then clone it locally:
```

`git clone https://github.com/<your Github username>/community-events.git`
and edit to your liking.

## Training on your own data
You can of course also train on your own images. For this, one can leverage Datasets' ImageFolder. Make sure to authenticate with the hub first, by running the huggingface-cli login command in a terminal, or the following in case you're working in a notebook:

```
python
from huggingface_hub import notebook_login

notebook_login()
```
Next, run the following in a notebook/script:
```
python
from datasets import load_dataset

# first: load dataset
# option 1: from local folder
dataset = load_dataset("imagefolder", data_dir="path_to_folder")
# option 2: from remote URL (e.g. a zip file)
dataset = load_dataset("imagefolder", data_files="URL to .zip file")

# next: push to the hub (assuming git-LFS is installed)
dataset.push_to_hub("huggan/my-awesome-dataset")
# You can then simply pass the name of the dataset to the script:

python train.py --dataset huggan/my-awesome-dataset
```
## Pushing model to the Hub

> For this you can use `push_to_hub_keras` which generates a card for your model with training metrics, plot of the architecture and hyperparameters. For this, specify --output_dir and --model_name and use --push_to_hub flag like so:

Reviewer: ```suggestion
For this you can use `push_to_hub_keras` which generates a card for your model with training metrics, plot of the architecture and hyperparameters. For this, specify `--output_dir` and `--model_name` and use the `--push_to_hub` flag like so:
```
